### PR TITLE
streamlit 1.37.1

### PR DESCRIPTION
**Pull request by psteyer**
streamlit 1.37.1

**Destination channel:** Defaults

### Links

- [PKG-5416]
- dev_url:        https://github.com/streamlit/streamlit/tree/release/1.37.1
- conda_forge:    None
- pypi:           https://pypi.org/project/streamlit/1.37.0
- pypi inspector: https://inspector.pypi.io/project/streamlit/1.37.0/
- related PR: https://github.com/AnacondaRecipes/st-theme-feedstock/pull/1
                     https://github.com/AnacondaRecipes/streamlit-navigation-bar-feedstock/pull/1

### Explanation of changes:

- updated version and dependency pinnings
- added python constraint
    - https://github.com/streamlit/streamlit/blob/200f5cfca44d19176972bd9ecb414bfc8b411841/lib/setup.py#L143

[PKG-5416]: https://anaconda.atlassian.net/browse/PKG-5416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ifitchet** [comment]
There's a Python restriction, `!=3.9.7`: https://inspector.pypi.io/project/streamlit/1.37.0/packages/68/b1/3fd1ae1e313f819a9e57272bd820130dd38e43bcbe9846ceba89345d45a6/streamlit-1.37.0.tar.gz/streamlit-1.37.0/setup.py#line.143

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
-  sha256: 65bf22e190de973b910c0b127b33d80d31c5275ed891c6fde740de46e5e05323
+  sha256: bc7e3813d94a39dda56f15678437eb37830973c601e8e574f2225a7bf188ea5a
 
 build:
   number: 0
   script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
   entry_points:
     - streamlit = streamlit.web.cli:main
-  skip: true  # [s390x or py<38]
+  skip: true  # [s390x or py<=38]
```
Comment (ifitchet) [inline recipe/meta.yaml:17]: Based on `Requires-Python: >=3.8, !=3.9.7`?:

```suggestion
  skip: true  # [s390x or py<38]
```

**Review by ifitchet — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,27 +19,27 @@ build:
...
     - requests >=2.27,<3
     - rich >=10.14.0,<14
     - tenacity >=8.1.0,<9
     - toml >=0.10.1,<2
     - typing_extensions >=4.3.0,<5
-    - watchdog >=2.1.5  # [not osx]
+    - watchdog >=2.1.5,<5  # [not osx]
     - gitpython >=3.0.7,<4,!=3.1.19
     - pydeck >=0.8.0,<1
     - tornado >=6.0.3,<7
```
Comment (Jrice1317) [inline recipe/meta.yaml:45]:
```suggestion
    - tornado >=6.0.3,<7
    run_constrained:
    - snowflake-snowpark-python >=0.9.0 # [py<312]
    - snowflake-connector-python >=2.8.0 # [py<312]
```

Comment (Jrice1317) [reply recipe/meta.yaml:45]: Maybe we should add [these](https://github.com/streamlit/streamlit/blob/1.37.1/lib/setup.py#L72-L77) to run_constrained?

**Review by Jrice1317 — COMMENTED** [review]
Outside of adding the extra requires with pinnings to run constrained, LGTM

**Review by Jrice1317 — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (psteyer) [reply recipe/meta.yaml:45]: added, and moved python skip to `run_constrained` also. Thank you!

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by ifitchet — APPROVED** [review]
Huh, the old `run_constrained` trick.

**Pull request merged**